src/chocolatey.tests.integration/scenarios/UpgradeAllScenarios.cs

```diff
@@ -0,0 +1,105 @@
+
+            public override void Context()
+            {
+                Configuration = Scenario.upgrade();
+                Scenario.reset(Configuration);
+                Configuration.PackageNames = Configuration.Input = "all";
+                Scenario.add_packages_to_source_location(Configuration, "upgradepackage*" + Constants.PackageExtension);
+                Scenario.add_packages_to_source_location(Configuration, "installpackage*" + Constants.PackageExtension);
+                Scenario.install_package(Configuration, "installpackage", "1.0.0");
+                Scenario.install_package(Configuration, "upgradepackage", "1.0.0");
+                Configuration.SkipPackageInstallProvider = true;
+
+                Service = NUnitSetup.Container.GetInstance<IChocolateyPackageService>();
+            }
+        }
+
+        [Concern(typeof (ChocolateyUpgradeCommand))]
+        public class when_upgrading_all_packages_happy_path : ScenariosBase
+        {
+            public override void Because()
+            {
+                Results = Service.upgrade_run(Configuration);
+            }
+
+            [Fact]
+            public void should_report_for_all_installed_packages()
+            {
+                Results.Count().ShouldEqual(2);
+            }
+
+            [Fact]
+            public void should_upgrade_packages_with_upgrades()
+            {
+                var upgradePackageResult = Results.Where(x => x.Key == "upgradepackage").ToList();
+                upgradePackageResult.Count.ShouldEqual(1, "upgradepackage must be there once");
+                upgradePackageResult.First().Value.Version.ShouldEqual("1.1.0");
+            }
+
+            [Fact]
+            public void should_skip_packages_without_upgrades()
+            {
+                var installPackageResult = Results.Where(x => x.Key == "installpackage").ToList();
+                installPackageResult.Count.ShouldEqual(1, "installpackage must be there once");
+                installPackageResult.First().Value.Version.ShouldEqual("1.0.0");
+            }
+        }
+
+        [Concern(typeof(ChocolateyUpgradeCommand))]
+        public class when_upgrading_all_packages_with_except : ScenariosBase
+        {
+            public override void Because()
+            {
+                Configuration.UpgradeCommand.PackageNamesToSkip = "upgradepackage";
+                Results = Service.upgrade_run(Configuration);
+            }
+
+            [Fact]
+            public void should_report_for_all_non_skipped_packages()
+            {
+                Results.Count().ShouldEqual(1);
+            }
+
+            [Fact]
+            public void should_skip_packages_in_except_list()
+            {
+                var upgradePackageResult = Results.Where(x => x.Key == "upgradepackage").ToList();
+                upgradePackageResult.Count.ShouldEqual(0, "upgradepackage should not be in the results list");
+            }
+        }
+    }
+}
```

Comment:
> Also, the upgrade tests all shared a common "context" (test setup), so changing that would require refactoring the whole UpgradeScenarios class, changing each test or restructuring them... but the PR only focus on filtering choco upgrade all packages, so changing this file completely - adding to this eventual merge issues if the PR is not quickly merged - falls outside of the actual except feature.


Well they share it, but they can override it in each individual spec class. And if you look through many of them, they do. Some override and still call the base, although not required. The way it is setup is quite flexible, no refactoring required.